### PR TITLE
chore(trusted.ci.jio/updates.jio) simplify and rename outputs

### PR DESCRIPTION
**Pull request by dduportal**
Related to https://github.com/jenkins-infra/helpdesk/issues/2649 and blocking https://github.com/jenkins-infra/update-center2/pull/784 and https://github.com/jenkins-infra/charts-secrets/pull/30

This PR simplifies and rename outputs to ensure the script used to generate the Update Center credential ZIP file in https://github.com/jenkins-infra/charts-secrets/pull/30 can extract the proper output with the naming convention from https://github.com/jenkins-infra/update-center2/pull/777 and https://github.com/jenkins-infra/update-center2/pull/784 (`www-content` and `www-redirections`).

- It avoids multiple levels of escaping challenges (current output requires 3 levels of escaping: in HCL syntax AND in shell syntax for the initial call AND on shell syntax in the file template itself)
- The only "coupling" in Terraform HCL is the output name which is an acceptable contract: it avoid having to change things in the terraform state when only renaming a variable
- No more risk to have a local env file committed by error and exposing the secrets

### CODE REVIEW

File: scripts/get-updates.jenkins.io-trustedci-credentials.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  zip_elements+=("${env_file}")
+done
+# terraform output update_center_fileshare_env-redirections > "${temp_dir}/.env-redirections"
+
+pushd "${temp_dir}"
+zip update-center-fileshares-env-zip-credentials.zip "${zip_elements[@]}"
+rm -f "${zip_elements[@]}"
+popd
+
+echo "ZIP file with credentials available in: ${temp_dir}/update-center-fileshares-env-zip-credentials.zip"
```
Comment (hervelemeur) [inline scripts/get-updates.jenkins.io-trustedci-credentials.sh:23]:
```suggestion
echo "ZIP file with credentials available at: ${temp_dir}/update-center-fileshares-env-zip-credentials.zip"
```
(Nit)

**Review by hervelemeur — APPROVED** [review]
Nice!

File: trusted.ci.jenkins.io.tf [hunk 2]
```diff
@@ -111,33 +111,33 @@ output "trustedci_updates_jenkins_io_httpd_fileshare_serviceprincipal_writer_pas
...
-    echo "STORAGE_NAME=updatesjenkinsio"
-    echo "STORAGE_FILESHARE=updates-jenkins-io-httpd"
-    echo "FILESHARE_SYNC_SOURCE=./www-redirections/"
-    echo "JENKINS_INFRA_FILESHARE_CLIENT_ID=${module.trustedci_updates_jenkins_io_httpd_fileshare_serviceprincipal_writer.fileshare_serviceprincipal_writer_application_client_id}"
-    echo "JENKINS_INFRA_FILESHARE_CLIENT_SECRET=${module.trustedci_updates_jenkins_io_httpd_fileshare_serviceprincipal_writer.fileshare_serviceprincipal_writer_password}"
-    echo "STORAGE_DURATION_IN_MINUTE=5"
-    echo "STORAGE_PERMISSIONS=dlrw"
-  } > .env-redirections && zip update-center-fileshares-env-zip-credentials.zip .env-content .env-redirections && rm .env-content .env-redirections
-    ZIPCOMMAND
+  value     = <<-EOF
```
Comment (hervelemeur) [inline trusted.ci.jenkins.io.tf:117]: What's the purpose of the dash in front of "EOF"?

**Comment by dduportal** [comment]
> Nice!

Still a draft an might change a lot: coupling is still really high (variables expected in update-center2 are present in Terraform which forces us to change terraform state if adding/renaming/removing a variable).

File: trusted.ci.jenkins.io.tf (hunk 2 above)
Comment (dduportal) [reply trusted.ci.jenkins.io.tf:117]: https://developer.hashicorp.com/terraform/language/expressions/strings#indented-heredocs

File: scripts/get-updates.jenkins.io-trustedci-credentials.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+zip_elements=()
+
+for service in content redirections
+do
+  env_file=.env-"${service}"
+  # The -raw flag ensures there are not heredoc markers in the output - https://developer.hashicorp.com/terraform/cli/commands/output#raw
+  terraform output -raw -no-color update_center_fileshare_env-"${service}" > "${temp_dir}"/"${env_file}"
+  zip_elements+=("${env_file}")
+done
+# terraform output update_center_fileshare_env-redirections > "${temp_dir}/.env-redirections"
```
Comment (pilere) [inline scripts/get-updates.jenkins.io-trustedci-credentials.sh:16]:
```suggestion
```

Comment (smerle33) [reply scripts/get-updates.jenkins.io-trustedci-credentials.sh:16]: sorry for the wrong account but it's still me @smerle33

**Review by dduportal — COMMENTED** [review]
Self merging: only `Changes to Outputs:`

**Pull request merged**